tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,211 @@
+"""
+This file compares dunder methods of DictConfig (e.g. __getitem__ and
+__setitem__) with dunder methods of standard python's dict.
```

Comment:
comment is outdated.

Reply:
It is?

Reply:
It's testing more than dunder methods now, isn't it?

Reply:
Nope. Just `__getitem__`, `__setitem__`, `__eq__` and `__contains__`.

I just realized when typing this that it might be good to add some tests for `__delitem__` too...

Reply:
and maybe pop and get() :).

How are you accounting for the different behavior of __getitem__ if an item is not found?
This is one of my early mistakes, it returns None in that case.
At some point I would like to transition toward a standard behavior. Maybe after 2.1.

Reply:
This is the `__getitem__` pattern that I'm currently using in the `test_compare_dictconfig_vs_dict.py` file:
```python
        try:
            result = python_dict[key]
        except KeyError:
            assert cfg[key] is None
        else:
            assert result == cfg[key]
```

Reply:
Okay, once my #516 lands simplify this.
Can you also test see how things are behaving with and without struct mode? should probably be tested for all functions here).

Reply:
> and maybe pop and get() :).

Ok, will do!

> Okay, once my #516 lands simplify this.
Can you also test see how things are behaving with and without struct mode? should probably be tested for all functions here).

Yup.

Reply:
> Okay, once my #516 lands simplify this.
Unfortunately the no simpler now than it was before:
```python
        try:
            result = python_dict[key]
        except KeyError:
            with pytest.raises(ConfigKeyError):
                cfg[key]
        else:
            assert result == cfg[key]
```

Reply:
We now have tests for all of the following:
```
    __contains__
    __delitem__
    __eq__
    __getitem__
    __setitem__
    get
    pop
    keys
    values
    items
```

Reply:
I've added conditional branches that deal with different behavior for structured vs non-structured `DictConfig`.